From 38756397211ed5a76843b55b7a5109877178986b Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Thu, 15 Jun 2017 18:48:22 +0300
Subject: [PATCH 1968/5242] MLK-14663-2: ASoC: fsl: imx-wm8960: Kill warning
 for non-gpr boards

commit  8dcd35ecce49427c95eabffaed43bd055455c947 from
https://source.codeaurora.org/external/imx/linux-imx.git

A side effect of commit 5555277e693a7 ("MLK-13574-1: ASoC: imx-wm8960:
remove the gpr dependency") is that a warning was printed for boards
without gpr. This can be confusing.

imx7d boards do not have a gpr setting, so use imx7d-evk-wm8960
compatible string to avoid printing the warning.

Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-wm8960.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/fsl/imx-wm8960.c b/sound/soc/fsl/imx-wm8960.c
index d3413c8..b905f2c 100644
--- a/sound/soc/fsl/imx-wm8960.c
+++ b/sound/soc/fsl/imx-wm8960.c
@@ -427,6 +427,10 @@ static int of_parse_gpr(struct platform_device *pdev,
 	int ret;
 	struct of_phandle_args args;
 
+	if (of_device_is_compatible(pdev->dev.of_node,
+				    "fsl,imx7d-evk-wm8960"))
+		return 0;
+
 	ret = of_parse_phandle_with_fixed_args(pdev->dev.of_node,
 					       "gpr", 3, 0, &args);
 	if (ret) {
@@ -649,6 +653,7 @@ static int imx_wm8960_remove(struct platform_device *pdev)
 
 static const struct of_device_id imx_wm8960_dt_ids[] = {
 	{ .compatible = "fsl,imx-audio-wm8960", },
+	{ .compatible = "fsl,imx7d-evk-wm8960"  },
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, imx_wm8960_dt_ids);
-- 
1.7.9.5

